### ISSUE TITLE
Corrupt bundles are not overwritten by `crc setup`

### ISSUE THREAD

**Issue by ajaysimha** [code-ready/crc#747]
### General information

  * OS: Linux (Fedora 30)
  * Hypervisor: KVM 
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```bash
# [asimha@asimha-fedora crc-linux-1.0.0-rc.0-amd64]$ crc version
version: 1.0.0-rc.0+34371d3
OpenShift version: 4.2.0-0.nightly-2019-09-26-192831 (embedded in binary)

```
  
## CRC status
```bash
# Put `crc status` output here
```

## CRC config
```bash
# Put `crc config view` output here
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
```
[asimha@asimha-fedora Downloads]$ cat /etc/redhat-release 
Fedora release 30 (Thirty)

### Steps to reproduce

  1. Download CRC and pull-secret.txt
  2. run crc setup
  3. run crc start
  4. 

### Expected
complete without errors

### Actual
[asimha@asimha-fedora crc-linux-1.0.0-rc.0-amd64]$ ./crc start -p ~/Downloads/pull-secret.txt 
INFO Checking if running as non-root              
INFO Checking if oc binary is cached              
INFO Checking if Virtualization is enabled        
INFO Checking if KVM is enabled                   
INFO Checking if libvirt is installed             
INFO Checking if user is part of libvirt group    
INFO Checking if libvirt is enabled               
INFO Checking if libvirt daemon is running        
INFO Checking if a supported libvirt version is installed 
INFO Checking if crc-driver-libvirt is installed  
INFO Checking if libvirt 'crc' network is available 
INFO Checking if libvirt 'crc' network is active  
INFO Checking if NetworkManager is installed      
INFO Checking if NetworkManager service is running 
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists 
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists 
INFO Extracting bundle: crc_libvirt_4.2.0-0.nightly-2019-09-26-192831.crcbundle ... 
ERRO Error occurred: Error getting bundle metadata: xz: data is truncated or corrupt 

### Logs

You can start crc with `crc start --log-level debug` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.
https://gist.github.com/ajaysimha/2dcbe27465e5e1c3d137e7be7fea80c7

**Comment by ajaysimha** [comment]
BTW I did sha256sum on my download:
[asimha@asimha-fedora Downloads]$ sha256sum -b ./crc-linux-amd64.tar.xz 
4c75bb69e536bbc1238933c8fea909a5270202c48c55c1e9ce1bee4d943c18e2 *./crc-linux-amd64.tar.xz

It matches https://mirror.openshift.com/pub/openshift-v4/clients/crc/latest/sha256sum.txt:
4c75bb69e536bbc1238933c8fea909a5270202c48c55c1e9ce1bee4d943c18e2  crc-linux-amd64.tar.xz

**Comment by gbraad** [comment]
How much memory in your computer?

**Comment by gbraad** [comment]
awkward issue, seems to be an incomplete slicing from the binary, and this causes the archive to be incomplete perhaps (and therefore corrupt)

**Comment by ajaysimha** [comment]
Hi,

I did "file crc_libvirt_4.2.0-0.nightly-2019-09-26-192831.crcbundle" and it showed me zero bytes.
After this, i did "rm -rf .crc"
Cleaned up
Ran "crc setup" again
After that it runs fine.

I don't know if you want to to pursue this any further or close the bug?

Thank you for your support!

Ajay

**Comment by ajaysimha** [comment]
May be this was a freak situation. However, I think If I downloaded new/fresh download I should have not seen this. I'd counter that 'crc setup' should wipe out .crc dir and start over. I'm closing this bug for now.

**Comment by ajaysimha** [comment]
Closing this bug as it is not longer an issue. If some does run into this issue, do a 'rm -rf .crc' and thn run 'crc setup' it resolved the issue for me.

**Comment by cfergeau** [comment]
> However, I think If I downloaded new/fresh download I should have not seen this. I'd counter that 'crc setup' should wipe out .crc dir and start over. I'm closing this bug for now.

crc should be robust against that situation, we should keep this open to track it.

**Comment by ajaysimha** [comment]
I agree. For a newbee like me. I have no idea. On a slow link I downloaded the package 3 times waiting all night. Better to take care of this during install and avoid unnecessary traffic from newbees

**Comment by gbraad** [comment]
We bundle code needs to be inspected, as to make space I also removed the bundle, but keeping the unpacked version. Due to this is started to complain (placed the file back) but is unused. We might need to record some of the edge cases here and resolve them. In the case described, we might need to check a checksum and decide if the file is OK... but only in the case when an extraction is really neeed (to avoid taking a penalty to check this).

**Comment by cfergeau** [comment]
We could also stop having separate steps between extracting bundle from the binary and uncompressing it, and do both as one step. So we would only be doing all of this when we don't have the directory we expect, and we would not need the intermediate bundle copy.

**Comment by cfergeau** [comment]
This branch https://github.com/cfergeau/crc/commits/embed has some changes which would help with that.

**Comment by akieling** [comment]
This issue is happening to me with the just released CRC [1.8.0+0a318dc](https://github.com/code-ready/crc/releases/tag/1.8.0).
Output below:
```
~ crc start --log-level debug
DEBU No new version available. The latest version is 1.8.0 
INFO Checking if oc binary is cached              
DEBU oc binary already cached                     
INFO Checking if podman remote binary is cached   
DEBU podman remote binary already cached          
INFO Checking if running as non-root              
INFO Checking if Virtualization is enabled        
DEBU Checking if the vmx/svm flags are present in /proc/cpuinfo 
DEBU CPU virtualization flags are good            
INFO Checking if KVM is enabled                   
DEBU Checking if /dev/kvm exists                  
DEBU /dev/kvm was found                           
INFO Checking if libvirt is installed             
DEBU Checking if 'virsh' is available             
DEBU 'virsh' was found in /usr/bin/virsh          
INFO Checking if user is part of libvirt group    
DEBU Checking if current user is part of the libvirt group 
DEBU Running '/usr/bin/groups akieling'           
DEBU Current user is already in the libvirt group 
INFO Checking if libvirt is enabled               
DEBU Checking if libvirtd.service is enabled      
DEBU Running '/usr/bin/systemctl is-enabled libvirtd' 
DEBU libvirtd.service is already enabled          
INFO Checking if libvirt daemon is running        
WARN Skipping above check ...                     
INFO Checking if a supported libvirt version is installed 
DEBU Checking if libvirt version is >=3.4.0       
DEBU Running 'virsh -v'                           
INFO Checking if crc-driver-libvirt is installed  
DEBU Checking if crc-driver-libvirt is installed  
DEBU Running '/home/akieling/.crc/bin/crc-driver-libvirt version' 
DEBU crc-driver-libvirt is already installed in /home/akieling/.crc/bin/crc-driver-libvirt 
INFO Checking if libvirt 'crc' network is available 
DEBU Checking if libvirt 'crc' network exists     
DEBU Running 'virsh --connect qemu:///system net-info crc' 
DEBU Checking if libvirt 'crc' definition is up to date 
DEBU Running 'virsh --connect qemu:///system net-dumpxml --inactive crc' 
DEBU libvirt 'crc' network has the expected value 
INFO Checking if libvirt 'crc' network is active  
DEBU Checking if libvirt 'crc' network is active  
DEBU Running 'virsh --connect qemu:///system net-info crc' 
DEBU libvirt 'crc' network is already active      
INFO Checking if NetworkManager is installed      
DEBU Checking if 'nmcli' is available             
DEBU 'nmcli' was found in /usr/bin/nmcli          
INFO Checking if NetworkManager service is running 
DEBU Checking if NetworkManager.service is running 
DEBU Running '/usr/bin/systemctl is-active NetworkManager' 
DEBU NetworkManager.service is already running    
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists 
DEBU Checking NetworkManager configuration        
DEBU NetworkManager configuration is good         
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists 
DEBU Checking dnsmasq configuration               
DEBU dnsmasq configuration is good                
Checking file: /home/akieling/.crc/machines/crc/.crc-exist
? Image pull secret [? for help] INFO Extracting bundle: crc_libvirt_4.3.8.crcbundle ... 
DEBU Uncompressing /home/akieling/.crc/cache/crc_libvirt_4.3.8.crcbundle to /home/akieling/.crc/cache 
ERRO Error getting bundle metadata: xz: data is corrupt 
```

**Comment by gbraad** [comment]
We added partial checksum and size checks...

... I'd rather create a new issue of a regression occurs.
